modules/graphql_core/src/BatchedFieldResolver.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Drupal\graphql_core;
+
+use Drupal\graphql_core\GraphQL\BatchedFieldInterface;
+
+/**
+ * Queueing service for deferred field resolution.
+ */
+class BatchedFieldResolver {
+  protected $queue;
+
+  /**
+   * Add a new field evaluation to the queue.
+   *
+   * @param \Drupal\graphql_core\GraphQL\BatchedFieldInterface $batchedField
+   *   The field instance.
+   * @param mixed $value
+   *   The parent value.
+   * @param array $args
+   *   The arguments it has been invoked with.
+   *
+   * @return array
+   *   A ticket that has to be used to retrieve the evaluation result.
+   */
+  public function enqueue(BatchedFieldInterface $batchedField, $value, array $args) {
+    $id = $batchedField->getBatchId($value, $args);
+    $this->queue[$id][] = [
+      'parent' => $value,
+      'arguments' => $args,
+      'field' => $batchedField,
+    ];
+    return [$id, max(array_keys($this->queue[$id]))];
+  }
+
+  /**
+   * Retrieve a result for a specific ticket.
+   *
+   * @param array $key
+   *   The deferred evaluation ticket.
+   *
+   * @return mixed
+   *   The evaluation result.
+   *
+   * @throws \Exception
+   *   In case the ticket is not valid any more.
+   */
+  public function resolve(array $key) {
```

Comment:
If we use a callable for this, we should change this method to use 2 explicit args instead of an array.